### ISSUE TITLE
[Bug]: mc crashes with message Bad system call

### ISSUE THREAD

**Issue by IgorSimic** [termux/termux-packages#14117]
### Problem description

mc crashes with message Bad system call.
Mc was working on this tablet but last today I noticed this issue.
What is changed in the mean time I really don't know, I regullary use pkg upgrade.
versions:
```
u0_a146@localhost:~$ pkg info mc
Package: mc
Version: 4.8.28
Maintainer: @termux
Installed-Size: 2716 kB
Depends: glib, libandroid-support, libssh2, ncurses
Homepage: https://www.midnight-commander.org/
Download-Size: 563 kB
APT-Manual-Installed: yes
APT-Sources: https://termux.cdn.lumito.net/termux-main stable/main arm Packages
Description: Midnight Commander - a powerful file manager
```

### What steps will reproduce the bug?

Just run it in same environment.

### System information

termux-info:

```
Termux Variables:
TERMUX_APP_PACKAGE_MANAGER=apt
TERMUX_MAIN_PACKAGE_FORMAT=debian
TERMUX_VERSION=0.118.0
Packages CPU architecture:
arm
Subscribed repositories:
# sources.list
deb https://termux.cdn.lumito.net/termux-main stable main
Updatable packages:
All packages up to date
termux-tools version:
1.34.1
Android version:
8.1.0
Kernel build information:
Linux localhost 3.18.14-15260192 #1 SMP PREEMPT Tue Nov 3 16:08:25 KST 2020 armv8l Android
Device manufacturer:
samsung
Device model:
SM-T580
```

**Comment by 2096779623** [comment]
I can't replicate this problem on aarch64 devices. Does it only happen on the arm architecture?

```bash
~ $ mc

~ $ 
```

**Comment by IgorSimic** [comment]
Yes, only on arm architecture.

---
Igor Simic
---
________________________________

**Comment by IgorSimic** [comment]
I think that problem starts wit updated libandroid-support package, but I am not sure.
```
u0_a146@localhost:~$ pkg info libandroid-support
Package: libandroid-support
Version: 28-3
Essential: yes
Maintainer: @termux
Installed-Size: 73.7 kB
Homepage: https://github.com/termux/libandroid-support
Download-Size: 10.2 kB
APT-Manual-Installed: yes
APT-Sources: https://termux.librehat.com/apt/termux-main stable/main arm Packages
Description: Library extending the Android C library (Bionic) for additional multibyte, locale and math support
```

**Comment by xtkoba** [comment]
Confirmed. Excerpt from `strace` output:

```
[...]
fstatat64(AT_FDCWD, "/data/data/com.termux/files/home/.local/share/mc", 0xff985eb8, 0) = 0
faccessat(AT_FDCWD, "/data/data/com.termux/files/home/.local/share/mc", F_OK) = 0
openat(AT_FDCWD, "/data/data/com.termux/files/usr/etc/mc/sfs.ini", O_RDONLY|O_LARGEFILE) = 3
fstat64(3, 0xff985d60)                  = 0
read(3, "#\n# This is config for Single Fi"..., 4096) = 871
read(3, "", 4096)                       = 0
close(3)                                = 0
openat(AT_FDCWD, "/data/data/com.termux/files/home/.local/share/mc/extfs.d", O_RDONLY|O_LARGEFILE|O_CLOEXEC|O_DIRECTORY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/data/data/com.termux/files/usr/lib/charset.alias", O_RDONLY|O_LARGEFILE) = -1 ENOENT (No such file or directory)
futex_time64(0xef366848, FUTEX_WAKE_PRIVATE, 2147483647) = 4013320264
--- SIGSYS {si_signo=SIGSYS, si_code=SYS_SECCOMP, si_call_addr=0xef123dc8, si_syscall=__NR_futex_time64, si_arch=AUDIT_ARCH_ARM} ---
[...]
```

**Comment by xtkoba** [comment]
From `logcat` output:

```
 *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
 Build fingerprint: 'asus/WW_X01AD/ASUS_X01A_1:8.1.0/OPM1/15.2016.1902.159-20190227:user/release-keys'
 Revision: '0'
 ABI: 'arm'
 pid: 6827, tid: 6827, name: mc  >>> mc <<<
 signal 31 (SIGSYS), code 1 (SYS_SECCOMP), fault addr --------
 Cause: seccomp prevented call to disallowed arm system call 422
     r0 e79bd848  r1 00000081  r2 7fffffff  r3 00000000
     r4 00000000  r5 e79bd3b4  r6 e79bd844  r7 000001a6
     r8 00000000  r9 00000002  sl 00000004  fp fffce768
     ip fffcd598  sp fffcd588  lr e79b468d  pc e7671dc8  cpsr 20010010

 backtrace:
     #00 pc 00018dc8  /system/lib/libc.so (syscall+32)
     #01 pc 00054689  /data/data/com.termux/files/usr/lib/libglib-2.0.so (offset 0x7e000)
```

I suppose `glib` (version `2.74.4`) should be to blame. Not `libandroid-support`.

**Comment by IgorSimic** [comment]
I am not developer, but can I do something about?

**Comment by xtkoba** [comment]
This bug report itself is quite helpful. Thanks for your contribution.

**Comment by xtkoba** [comment]
Should be fixed with `glib` version `2.74.4-1` which will soon become available.

**Comment by IgorSimic** [comment]
I can confirm, updated glib 2.74.4-1 solved the problem.
